File: LinkedList/Remove Nth Node From End of List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        
        ListNode *fast = head, *slow = head;
        
        // Fast pointer will be ahead by n steps
        for(int i = 0; i < n; i++){
            fast = fast->next;
        }
        // If 1st element needs to be deleted
        if(fast == nullptr){
            head = head->next;
            return head;
        }
        // Else let the pointers reach the end of the list
        else{
            while(fast->next != nullptr){
                fast = fast->next;
                slow = slow->next;
            }

            slow->next = slow->next->next;
            return head;
        }
    }
};

File: LinkedList/Remove Nth Node From End of List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 
func removeNthFromEnd(head *ListNode, n int) *ListNode {
    var curr *ListNode = head;
    var prev *ListNode = nil;

    var end *ListNode = head;
    for i := 1; i < n; i++ {
        end = end.Next;
    }

    for end.Next != nil {
        prev = curr;
        end = end.Next;
        curr = curr.Next;
    }

    if end == head {
        return head.Next;
    } else if prev == nil {
        return curr.Next;
    } else {
        prev.Next = curr.Next;
        return head;
    }
}